### PR TITLE
Node: Add tests for invalid messages

### PR DESCRIPTION
**Pull request by bruce-riley**
Add tests for the code that checks for invalid chain / zero address in messages. Move the code into a separate function to facilitate testing.

### CODE REVIEW

**Comment by bruce-riley** [comment]
@hendrikhofstadt, I added code so that posting a message to the process won't block. But now I'm not so sure that's such a good idea. Do you think the watcher should hang until the processor can accept the observation, or should the message just get dropped (like I have it in this PR)?

File: node/cmd/guardiand/validate_msg_test.go
Comment (hendrikhofstadt) [reply node/cmd/guardiand/validate_msg_test.go:37]: Also, there is a shortcut require.True

File: node/cmd/guardiand/validate_msg_test.go
Comment (hendrikhofstadt) [inline node/cmd/guardiand/validate_msg_test.go:59]: require, see above

File: node/cmd/guardiand/node.go
Comment (panoel) [inline node/cmd/guardiand/node.go:1569]: Why not put this inside the function validateMessage()?

File: node/cmd/guardiand/validate_msg_test.go
Comment (panoel) [inline node/cmd/guardiand/validate_msg_test.go:41]: Why is this one different than the other 2 tests?

**Review by panoel — APPROVED** [review]

File: node/cmd/guardiand/node.go
Comment (bruce-riley) [reply node/cmd/guardiand/node.go:1569]: It's at least some overhead to set it to zero. Don't think it hurts to have it global.

**Review by hendrikhofstadt — DISMISSED** [review]

**Review by panoel — APPROVED** [review]

File: node/cmd/guardiand/node.go
Comment (tbjump) [inline node/cmd/guardiand/node.go:922]: I think the code becomes less readable this way, because now there are three different paths in which errors are handled 
1. by setting shouldPublish to false
2. by returning an error
3. by logging an error within validateMessage itself

**Review by tbjump — CHANGES_REQUESTED** [review]
I appreciate the spirit of adding unit tests for security checks like this one. But I think it'd be better to instead do the test on a higher level, i.e. inject a wrong message into a watcher channel and assert that it won't get published. That way we'd check not only that validateMessage is working correctly, but also that it is being used in the right place. 

I am working on a PR to refactor the node.go that will allow you to create a guardian within a unit test and then simulate a watcher (https://github.com/wormhole-foundation/wormhole/pull/2561). I'd recommend we wait until that PR is merged and then add these tests on top there.

**Pull request closed without merging**
